### PR TITLE
Streampark flink job package includes cliassifier packages

### LINKED ISSUE #2380: [Feature] When the flink job depends on the customized classifier package, the publishing package contains the classifier package
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

During the construction of Flink sql job, the classifier dependency package cannot be extracted from maven pom or uploaded.
Dependency like:
   &lt;dependency&gt;
       &lt;groupId&gt;org.apache.flink.dfsk&lt;/groupId&gt;
       &lt;artifactId&gt;flink-dfsk-sql-connector-emq&lt;/artifactId&gt;
       &lt;version&gt;1.14.6-SNAPSHOT&lt;/version&gt;
       &lt;classifier&gt;shade&lt;/classifier&gt;
   &lt;/dependency&gt;
or 
  flink-dfsk-sql-connector-emq-1.14.6-SNAPSHOT-shade.jar

### Usage Scenario

Upload the customized classifier package to the maven private server or  upload the jar. The compilation and packaging of the flink job needs to include the classifier package

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhangxiangyang**
## What changes were proposed in this pull request

Issue Number: close #2380  

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

File: streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue [hunk 1]
```diff
@@ -113,12 +116,15 @@
             }
           }
           if (invalidArtifact.length === 0) {
-            const id = groupId + '_' + artifactId;
+            const id = classifier != null ? groupId + '_' + artifactId + '_' + classifier : groupId + '_' + artifactId;
```
Comment (wolfboys) [inline streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue]: I suggest add a new method for this

File: streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue [hunk 2]
```diff
@@ -220,7 +226,7 @@
     handleUpdateDependency();
   }
   function handleRemovePom(pom: Recordable) {
-    const id = pom.groupId + '_' + pom.artifactId;
+    const id = pom.classifier != null ? pom.groupId + '_' + pom.artifactId + '_' + pom.classifier : pom.groupId + '_' + pom.artifactId;
```
Comment (wolfboys) [inline streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue]: repeated logic

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java [hunk 3]
```diff
@@ -699,17 +706,24 @@ public boolean equals(Object o) {
...
     @Override
     public String toString() {
-      return groupId + ":" + artifactId + ":" + version;
+      return groupId + ":" + artifactId + ":" + version + ":" + classifier;
     }
 
     @JsonIgnore
     public String getPath() {
-      return getGroupId() + "_" + getArtifactId() + "-" + getVersion() + ".jar";
+      return getGroupId()
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java]: if classifier is null, Will there be an exception?

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java [hunk 4]
```diff
@@ -685,6 +691,7 @@ public static class Pom {
     private String groupId;
     private String artifactId;
     private String version;
+    private String classifier;
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java:695]: The user's exist historical jobs. Without this classifier field, Will there be any exceptions?

File: streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue (hunk 1 above)
Comment (zhangxiangyang) [reply streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue]: Add new function 
function getId(groupId: string, artifactId: string, classifier: string) {
    return classifier != null ? groupId + '_' + artifactId + '_' + classifier : groupId + '_' + artifactId;
  }

File: streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue (hunk 2 above)
Comment (zhangxiangyang) [reply streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue]: Add new function 
function getId(groupId: string, artifactId: string, classifier: string) {
    return classifier != null ? groupId + '_' + artifactId + '_' + classifier : groupId + '_' + artifactId;
  }

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java (hunk 3 above)
Comment (zhangxiangyang) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java]: Add null value judgment

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java (hunk 4 above)
Comment (zhangxiangyang) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java:695]: Compile with the modified version, edit a new job, upload the classifier and non-classifier packages, the classifier and non-classifier POM, and compile the package publishing task normally.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java [hunk 5]
```diff
@@ -700,17 +707,28 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-      return Objects.hash(groupId, artifactId, version);
+      return Objects.hash(groupId, artifactId, version, classifier);
     }
 
     @Override
     public String toString() {
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java:714]: how about: 
```
 @Override
        public String toString() {
            return groupId
                + ":"
                + artifactId
                + ":"
                + version
                + getClassifier (":");
        }

        @JsonIgnore
        public String getPath() {
            return getGroupId()
                + "_"
                + getArtifactId()
                + "-"
                + getVersion()
                + getClassifier("_")
                + ".jar";
        }

        private String getClassifier(String joiner) {
            return StringUtils.isEmpty(classifier) ? "" : joiner + classifier;
        }
```

File: streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue [hunk 6]
```diff
@@ -250,6 +258,11 @@
     delete dependency.jar[item];
     handleUpdateDependency();
   }
+
+  function getId(groupId: string, artifactId: string, classifier: string) {
```
Comment (wolfboys) [inline streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue]:
```
function getId(pom) {
   if (pom.classifier != null ) {
     return pom.groupId + '_' + pom.artifactId + '_' + pom.classifier
   }
   return pom.groupId + '_' + pom.artifactId;
 }
```

File: streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue [hunk 7]
```diff
@@ -234,8 +240,10 @@
   function setDefaultValue(dataSource: { pom?: DependencyType[]; jar?: string[] }) {
     dependencyRecords.value = dataSource.pom || [];
     uploadJars.value = dataSource.jar || [];
+    dependency.pom = {}
+    dependency.jar = {}
     dataSource.pom?.map((pomRecord: DependencyType) => {
-      const id = pomRecord.groupId + '_' + pomRecord.artifactId;
+      const id = pomRecord.classifier != null ? pomRecord.groupId + '_' + pomRecord.artifactId + '_' + pomRecord.classifier : pomRecord.groupId + '_' + pomRecord.artifactId;
```
Comment (wolfboys) [inline streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue]: const id = getId(pomRecord);

File: streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue [hunk 8]
```diff
@@ -220,7 +226,7 @@
     handleUpdateDependency();
   }
   function handleRemovePom(pom: Recordable) {
-    const id = pom.groupId + '_' + pom.artifactId;
+    const id = getId(pom.groupId, pom.artifactId, pom.classifier);
```
Comment (wolfboys) [inline streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue]: const id = getId(pom);

File: streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue [hunk 9]
```diff
@@ -113,12 +116,15 @@
...
-            const id = groupId + '_' + artifactId;
+            const id = getId(groupId, artifactId, classifier);
             const mvnPom: Recordable = {
               groupId: groupId,
               artifactId: artifactId,
               version: version,
             };
+            if (classifier != null) {
+              mvnPom.classifier = classifier;
+            }
```
Comment (wolfboys) [inline streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue:126]: const id = getId(mvnPom);

Comment (zhangxiangyang) [reply streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue:126]: ok

File: streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue (hunk 8 above)
Comment (zhangxiangyang) [reply streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue]: ok

File: streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue (hunk 7 above)
Comment (zhangxiangyang) [reply streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue]: ok

File: streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue (hunk 6 above)
Comment (zhangxiangyang) [reply streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue]: ok

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java (hunk 5 above)
Comment (zhangxiangyang) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java:714]: ok

File: streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue [hunk 10]
```diff
@@ -314,9 +330,19 @@
       @click="handleEditPom(dept)"
     >
       <template #message>
-        <Space @click="handleEditPom(dept)" class="tag-dependency-pom">
+        <Space @click="handleEditPom(dept)" class="tag-dependency-pom" v-if="dept.classifier">
```
Comment (wolfboys) [inline streampark-console/streampark-console-webapp/src/views/flink/app/components/Dependency.vue]: how about? 
```
<Space @click="handleEditPom(dept)" class="tag-dependency-pom">
        <Tag class="tag-dependency" color="#2db7f5">POM</Tag>
        <template v-if="dept.classifier != null">
            {{ dept.artifactId }}-{{ dept.version }}-{{ dept.classifier }}.jar
        </template>
        <template v-else>
            {{ dept.artifactId }}-{{ dept.version }}.jar
        </template>
        <Icon
        :size="12"
        icon="ant-design:close-outlined"
        class="icon-close cursor-pointer"
        @click.stop="handleRemovePom(dept)"
        />
    </Space>
```

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java [hunk 11]
```diff
@@ -700,17 +707,27 @@ public boolean equals(Object o) {
...
-      return groupId + ":" + artifactId + ":" + version;
+      return groupId + ":" + artifactId + ":" + version + getClassifier(":");
     }
 
     @JsonIgnore
     public String getPath() {
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/Application.java]: unused method, we can delete this method

File: streampark-flink/streampark-flink-packer/src/main/scala/org/apache/streampark/flink/packer/maven/Artifact.scala [hunk 12]
```diff
@@ -21,7 +21,7 @@ import java.util.regex.Pattern
 
 import org.eclipse.aether.artifact.{Artifact => AetherArtifact}
 
-case class Artifact(groupId: String, artifactId: String, version: String) {
+case class Artifact(groupId: String, artifactId: String, version: String, classifier: String) {
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-packer/src/main/scala/org/apache/streampark/flink/packer/maven/Artifact.scala]:
```
case class Artifact(groupId: String, artifactId: String, version: String, classifier: String = null)
```

File: streampark-flink/streampark-flink-packer/src/main/scala/org/apache/streampark/flink/packer/maven/Artifact.scala [hunk 13]
```diff
@@ -49,7 +49,7 @@ object Artifact {
         val g = m.group(1)
         val a = m.group(2)
         val v = m.group(3)
-        Artifact(g, a, v)
+        Artifact(g, a, v, null)
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-packer/src/main/scala/org/apache/streampark/flink/packer/maven/Artifact.scala]: `Artifact(g, a, v)`

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**